catroid/src/main/java/org/catrobat/catroid/stage/StageListener.java

```diff
@@ -435,20 +436,26 @@ public void reloadProject(StageDialog stageDialog) {
 			return;
 		}
 		this.stageDialog = stageDialog;
-		if (!ProjectManager.getInstance().getStartScene().getName().equals(scene.getName())) {
-			transitionToScene(ProjectManager.getInstance().getStartScene().getName());
+		ProjectManager projectManager = ProjectManager.getInstance();
+		Project currentProject = projectManager.getCurrentProject();
+
+		if (!projectManager.getStartScene().getName().equals(scene.getName())) {
+			transitionToScene(projectManager.getStartScene().getName());
 		}
 		stageBackupMap.clear();
 		embroideryPatternManager.clear();
 
+		if (currentProject.getRequiredResources().contains(Brick.VIBRATION)) {
+			VibrationUtil.reset();
```

Comment:
Is this check necessary? Getting all required resources is probably quite slow, so if resetting the time to 0 when vibration is not used doesn't affect anything, I would just leave out the conditional statement.

Reply:
Update:
- Killed "context" as a member in VibrationUtil
- Removed some time consuming if conditions


ad:
> Also, the StageResourceHolder already starts the thread. I would either eliminate that line so you don't create an unnecessary thread at the beginning or keep it and simply leave one vibration thread running and use a mutex instead of the binary semaphore to update the vibration time to tell the thread to start vibrating again. The latter option would probably cut down complexity quite a bit.

I will leave that as it is. Im not sure if there is an explicit mutex in java. There is some similar stuff like reentrantlocks, binary semaphores or synchronized objects, which behaves like mutual expressions, but i think the solution with the binary semaphore is just fine. Also the thread needs to get started at the beginning with the existing solution.